### DISCUSSION TITLE
Holiday Lights:  TypeError: Cannot read property 'state' of undefined from Date Ranges

### DISCUSSION THREAD

**Discussion by pvr2002** [zachowj/node-red-contrib-home-assistant-websocket#468] [Q&A] [answered]
Imported the node red flow into a new tab "Flow 3"
Updated the setup node entityid to "light.wled_greatroomcubby"
Updated the setup node IP to "192.168.200.195"
Had to update a handful of nodes with the correct server

Created the 3 helpers
- input_number.holiday_lights_demo_mode_delay
- input_select.holiday_lights_demo_mode
- input_select.holiday_lights_presets

The input select option sees all of the presets and I can change the preset from the dashboard, including Summer, Winter, Autumn, Spring.

The timer flow to turn off the WLED at midnight works great.
However the timer flow to turn on WLED is not working.  When I save the flow and look at the debug message I am getting the following message from the "date ranges" function:  TypeError: Cannot read property 'state' of undefined".

Any idea what I'm missing?

**Comment by zachowj** [comment]
Is the HA server config in NR named something other than `Home Assistant`?

If so, change line 10 to the value in the context menu under homeassistant

![image](https://user-images.githubusercontent.com/37859597/133867073-9b60e70e-3fa2-4c95-bfef-6be8f57181cc.png)

```
const states = global.get("homeassistant").homeAssistant.states;
```

would become

```
const states = global.get("homeassistant").localDev.states;
```

**Reply by pvr2002** [reply]
The HA server name in NodeRed is "Home Assistant."

**Comment by zachowj** [comment, marked as answer]
Do you have the season integration enabled in HA? 

https://www.home-assistant.io/integrations/season/

**Reply by pvr2002** [reply]
I believe I do.  I added the 2 lines from that page in my configuration.yaml and rebooted the system.

**Reply by zachowj** [reply]
Does `sensor.season` exist in HA?

**Reply by pvr2002** [reply]
I do see sensor.season under the developer tools.
![image](https://user-images.githubusercontent.com/86473755/133868634-db13d3f0-1934-4316-b115-a1b2e060f1f7.png)

**Reply by zachowj** [reply]
Then my only guess left would be that eztimer ran before NR had a chance to connect to HA. Is your deploy type set to Full so it resets the connection each time you deploy?

**Reply by pvr2002** [reply]
It is set to full when I deploy.  Out of curiosity, In the date ranges code, I see flow.get("wledHolidayLightsPresets") and flow.get("wledHolidayLightsEntityId").  I have been trying to figure out what exactly that is referring to.  I don't seem to find anything with those names.
![image](https://user-images.githubusercontent.com/86473755/133869377-13530562-253e-4136-95b1-2cdc3175c9aa.png)

**Reply by zachowj** [reply]
Try changing your deploy method to Modified Nodes.

Those are set in the `process data` function node from values in the `Setup` inject node and the `get presets` request node.

**Reply by pvr2002** [reply]
Looks like deploying it as "modified nodes" did the trick.  Thanks for the help.
